OpenVPNHelper/OpenVPNHelper.m

```diff
@@ -85,6 +85,43 @@ - (BOOL)verify:(NSString *)identifier atURL:(NSURL *)fileURL {
 }
 
 - (void)startOpenVPNAtURL:(NSURL *_Nonnull)launchURL withConfig:(NSURL *_Nonnull)config upScript:(NSURL *_Nullable)upScript downScript:(NSURL *_Nullable)downScript leasewatchPlist:(NSURL *_Nullable)leasewatchPlist leasewatchScript:(NSURL *_Nullable)leasewatchScript scriptOptions:(NSArray <NSString *>*_Nullable)scriptOptions reply:(void(^_Nonnull)(BOOL))reply {
+    
+    // Get the path of config file
+    NSString* path = config.path;
+    NSString* content = [NSString stringWithContentsOfFile:path
+                                                  encoding:NSUTF8StringEncoding
+                                                     error:NULL];
+    
+    NSMutableArray *listItems = [content componentsSeparatedByString:@"\r\n"];
+    NSArray *maliciousCommands = @[@"up", @"tls-verify", @"ipchange", @"client-connect", @"route-up",@"route-pre-down",@"client-disconnect",@"down",@"learn-address",@"auth-user-pass-verify"];
+    
+    
+    //loop through array to check if malicious is command
+    for ( int i = 0; i < [listItems count]; i++) {
+        NSString *line = [listItems objectAtIndex: i];
+        
+        
+        //Loop through malicious commands to check if any of them is present in the line
+        for ( int j = 0; j < [maliciousCommands count]; j++) {
+            NSString *maliciousCommand = [maliciousCommands objectAtIndex: j];
+            
+             //Check if malicious command was found
+            if ([line rangeOfString:[NSString stringWithFormat:@"%@%@", maliciousCommand,@" "]].location == NSNotFound) {
+                
+            } else {
+                syslog(LOG_NOTICE, "Malicious command removed");
+                [listItems removeObjectAtIndex: i];
```

Comment:
Changing an array while enumerating can crash the app or can cause a line of the config file to be skipped.

Reply:
So, what do you suggest?

Reply:
You'd collect the indices of the lines you want to remove and after you're done checking remove those.

But I suggest to just bail completely when any of these commands are present. That means returning an NO to the app. But we'd want more elaborate error reporting, so we'd probably need to change that to allow for a more detailed error to be sent to the app so that the user can be informed on which command that is disallowed.

Reply:
Good Point, Viscosity works that way. Those commands are useful of course for many company may be